src/smtp_socket.erl

```diff
@@ -120,14 +120,30 @@ accept(Socket, Timeout) when is_port(Socket) ->
 		Error ->
 			Error
 	end;
+
+
 accept(Socket, Timeout) ->
 	case ssl:transport_accept(Socket, Timeout) of
 		{ok, NewSocket} ->
-			ssl:ssl_accept(NewSocket),
+			ssl_handshake(NewSocket),
```

Comment:
in OTP 21, handshake always returns a new Socket and we are dropping it here ?

Reply:
Thanks, see https://github.com/gen-smtp/gen_smtp/pull/187 if that would solve this.  (All tests pass, so apparently at the moment the ssl routines return the same socket)